### PR TITLE
directory layout + configuration: next steps

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | #4052 
| License           | MIT
| Changelog updated | no

## Description

This PR includes the following changes:

- Update `src/Kernel.php` according [4.x recipe](https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/4.2/src/Kernel.php) and prepares for [5.1 recipe](https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.1/src/Kernel.php)
  - Introduces the usage of `MicroKernelTrait`
- Update `config/bootstrap.php` according [recipe](https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/4.2/config/bootstrap.php)
- Update `bin/console` according [recipe](https://github.com/symfony/recipes/blob/master/symfony/console/4.4/bin/console)
- Update `src/index.php` and `public/index.php` according [recipe](https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/4.4/public/index.php)
- Removed config files which now became unrequired (e.g. for loading routes)
- Renamed some other config files and eliminated `config*.yaml`
- Prepared usage of environment variables

### CODE REVIEW

**Comment by craigh** [comment]
travis build failing

**Comment by Guite** [comment]
I noticed that too... I wanted to wait until you got there to discuss it with you in Slack.

File: bin/console [hunk 1]
```diff
@@ -6,38 +6,43 @@ use Symfony\Component\Console\Input\ArgvInput;
...
 use Symfony\Component\Yaml\Yaml;
 
+if (!in_array(PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+    echo 'Warning: The console should be invoked via the CLI version of PHP, not the '.PHP_SAPI.' SAPI'.PHP_EOL;
+}
+
 set_time_limit(0);
 
-require __DIR__ . '/../vendor/autoload.php';
-require __DIR__ . '/../src/RequirementChecker.php';
```
Comment (craigh) [inline bin/console]: please return requirementChecker

File: bin/console [hunk 2]
```diff
@@ -6,38 +6,43 @@ use Symfony\Component\Console\Input\ArgvInput;
...
 
-// on install or upgrade, check if system requirements are met.
-$kernelConfig = Yaml::parse(file_get_contents(__DIR__ . '/../config/services.yaml'));
-if (is_readable($file = __DIR__ . '/../config/services_custom.yaml')) {
-    $kernelConfig = array_merge($kernelConfig, Yaml::parse(file_get_contents($file)));
+if ($input->hasParameterOption('--no-debug', true)) {
+    putenv('APP_DEBUG=' . $_SERVER['APP_DEBUG'] = $_ENV['APP_DEBUG'] = '0');
 }
-$requirementChecker = new RequirementChecker();
-$requirementChecker->verify($kernelConfig['parameters']);
```
Comment (craigh) [inline bin/console]: please return

File: bin/console (hunk 1 above)
Comment (Guite) [reply bin/console]: `console` includes `config/bootstrap.php` which includes and executes the `RequirementChecker`. No need to have it twice.

File: bin/console (hunk 2 above)
Comment (Guite) [reply bin/console]: `console` includes `config/bootstrap.php` which includes and executes the `RequirementChecker`. No need to have it twice.

File: config/bootstrap.php [hunk 3]
```diff
@@ -12,36 +12,47 @@
...
+use Symfony\Component\Dotenv\Dotenv;
 use Symfony\Component\Yaml\Yaml;
 
-require __DIR__ . '/../vendor/autoload.php';
-require __DIR__ . '/../src/RequirementChecker.php';
+require dirname(__DIR__) . '/vendor/autoload.php';
+require dirname(__DIR__) . '/src/RequirementChecker.php';
+
+// Load cached env vars if the .env.local.php file exists
+// Run "composer dump-env prod" to create it (requires symfony/flex >=1.2)
```
Comment (craigh) [inline config/bootstrap.php]: requried flex? :disappointed:

Comment (Guite) [reply config/bootstrap.php]: Supporting the option does not mean that we have to use Flex to assemble packages. Let's worry about other things first.

File: config/services.yaml [hunk 4]
```diff
@@ -48,6 +48,15 @@ parameters:
     # it accepts any values allowed for X-Frame-Options
     security.x_frame_options: SAMEORIGIN
 
+
+    #container.dumper.inline_class_loader: true
+
+    # all paths below are resolved as assets (in /web)
```
Comment (craigh) [inline config/services.yaml]: in `web`?

Comment (Guite) [reply config/services.yaml]: fixed in 27a361432cccea4b9ad72371414a893de64b079e

**Pull request merged**